modules/json/src/parsers/convert-json.js

```diff
@@ -108,73 +109,41 @@ export function getJSONLayers(jsonLayers = [], configuration) {
   const layerCatalog = configuration.layers || {};
   return jsonLayers.map(jsonLayer => {
     const Layer = layerCatalog[jsonLayer.type];
-    const props = getJSONLayerProps(jsonLayer, configuration);
+    const props = getJSONLayerProps(Layer, jsonLayer, configuration);
     props.fetch = enhancedFetch;
     return Layer && new Layer(props);
   });
 }
 
-function getJSONLayerProps(jsonProps, configuration) {
+function getJSONLayerProps(Layer, jsonProps, configuration) {
   const replacedProps = {};
   for (const propName in jsonProps) {
-    // eslint-disable-line guard-for-in
-    const propValue = jsonProps[propName];
-    // Handle accessors
-    if (propName.startsWith('get')) {
-      replacedProps[propName] = getJSONAccessor(propValue, configuration);
-    } else {
-      replacedProps[propName] = propValue;
-    }
-  }
-  return replacedProps;
-}
-
-// Calculates an accessor function from a JSON string
-// '-' : x => x
-// 'a.b.c': x => x.a.b.c
-function getJSONAccessor(propValue, configuration) {
-  if (propValue === '-') {
-    return object => object;
-  }
-  if (typeof propValue === 'string') {
-    return object => {
-      return get(object, propValue);
-    };
-  }
-  return propValue;
-}
-
-// HELPERS
-
-function enhancedFetch(url) {
-  /* global fetch */
-  return fetch(url)
-    .then(response => response.text())
-    .then(text => {
-      try {
-        return JSON.parse(text);
-      } catch (error) {
-        return parseCSV(text);
+    let propValue = jsonProps[propName];
+
+    // Parse string valued expressions
+    if (typeof propValue === 'string') {
+      const propType = Layer && Layer.defaultProps && Layer.defaultProps[propName];
```

Comment:
Prop type would not be in `Layer.defaultProps` if the layer is subclassed.

Reply:
I seem to recall that we calculate a `_mergedDefaultProps` field...

Reply:
> Prop type would not be in Layer.defaultProps if the layer is subclassed.

`Layer._propTypes` has the merged props from all the parent layers. We should probably create an official API to get these.

https://github.com/uber/deck.gl/blob/master/modules/core/src/lifecycle/create-props.js#L132